packages/jest-worker/src/workers/ChildProcessWorker.ts

```diff
@@ -44,19 +47,30 @@ import {
 export default class ChildProcessWorker implements WorkerInterface {
   private _child!: ChildProcess;
   private _options: WorkerOptions;
-  private _onProcessEnd!: OnEnd;
-  private _fakeStream: PassThrough | null;
+
   private _request: ChildMessage | null;
   private _retries!: number;
-  private _stderr: ReturnType<typeof mergeStream> | null;
+  private _onProcessEnd!: OnEnd;
+
+  private _fakeStream: PassThrough | null;
   private _stdout: ReturnType<typeof mergeStream> | null;
+  private _stderr: ReturnType<typeof mergeStream> | null;
```

Comment:
I know it's somewhat existing code style, but why `null` instead of just using TypeScript's `?` / `undefined`.

I've seen a bunch of places in the code where we end up with `undefined | null | Something` and I'd rather try to reserve `null` for special cases where it's actively un-setting something and default to `undefined` otherwise.

Not super necessary to refactor the world here but just wanted to call it out and see if there was a specific reason.

Reply:
I agree that for properties `undefined` is usually a better choice.
This diff is already too large to include more refactoring though 😅 just felt like structuring the class properties nicely here was starting to become necessary because they are quite many

Reply:
fair point :) maybe future todo